cuda_core/cuda/core/experimental/_memory.pyx

```diff
@@ -23,7 +23,28 @@
 """A type union of :obj:`~driver.CUdeviceptr`, `int` and `None` for hinting :attr:`Buffer.handle`."""
 
 
-class Buffer:
+cdef class _MembersNeededForFinalize:
```

Comment:
Please pay great attention to follow the PR #709 and the issue #658 that it tried to solve, in particular the last comment https://github.com/NVIDIA/cuda-python/issues/658#issuecomment-3020383113. This MNFF thing is known to be very slow due to the use of weakref finalizer. In 709 we just switched to make the destructor implemented as `__del__`.